packages/block-library/src/social-link/social-list.js

```diff
@@ -1,227 +1,31 @@
-	VkIcon,
-	WordPressIcon,
-	YelpIcon,
-	YouTubeIcon,
-} from './icons';
-
-const socialList = {
-	fivehundredpx: {
-		name: '500px',
-		icon: FivehundredpxIcon,
-	},
-	amazon: {
-		name: 'Amazon',
-		icon: AmazonIcon,
-	},
-	bandcamp: {
-		name: 'Bandcamp',
-		icon: BandcampIcon,
-	},
-	behance: {
-		name: 'Behance',
-		icon: BehanceIcon,
-	},
-	chain: {
-		name: 'Link',
-		icon: ChainIcon,
-	},
-	codepen: {
-		name: 'CodePen',
-		icon: CodepenIcon,
-	},
-	deviantart: {
-		name: 'DeviantArt',
-		icon: DeviantArtIcon,
-	},
-	dribbble: {
-		name: 'Dribbble',
-		icon: DribbbleIcon,
-	},
-	dropbox: {
-		name: 'Dropbox',
-		icon: DropboxIcon,
-	},
-	etsy: {
-		name: 'Etsy',
-		icon: EtsyIcon,
-	},
-	facebook: {
-		name: 'Facebook',
-		icon: FacebookIcon,
-	},
-	feed: {
-		name: 'RSS Feed',
-		icon: FeedIcon,
-	},
-	flickr: {
-		name: 'Flickr',
-		icon: FlickrIcon,
-	},
-	foursquare: {
-		name: 'Foursquare',
-		icon: FoursquareIcon,
-	},
-	goodreads: {
-		name: 'Goodreads',
-		icon: GoodreadsIcon,
-	},
-	google: {
-		name: 'Google',
-		icon: GoogleIcon,
-	},
-	github: {
-		name: 'GitHub',
-		icon: GitHubIcon,
-	},
-	instagram: {
-		name: 'Instagram',
-		icon: InstagramIcon,
-	},
-	lastfm: {
-		name: 'Last.fm',
-		icon: LastfmIcon,
-	},
-	linkedin: {
-		name: 'LinkedIn',
-		icon: LinkedInIcon,
-	},
-	mail: {
-		name: 'Mail',
-		icon: MailIcon,
-	},
-	mastodon: {
-		name: 'Mastodon',
-		icon: MastodonIcon,
-	},
-	meetup: {
-		name: 'Meetup',
-		icon: MeetupIcon,
-	},
-	medium: {
-		name: 'Medium',
-		icon: MediumIcon,
-	},
-	pinterest: {
-		name: 'Pinterest',
-		icon: PinterestIcon,
-	},
-	pocket: {
-		name: 'Pocket',
-		icon: PocketIcon,
-	},
-	reddit: {
-		name: 'Reddit',
-		icon: RedditIcon,
-	},
-	skype: {
-		name: 'Skype',
-		icon: SkypeIcon,
-	},
-	snapchat: {
-		name: 'Snapchat',
-		icon: SnapchatIcon,
-	},
-	soundcloud: {
-		name: 'SoundCloud',
-		icon: SoundCloudIcon,
-	},
-	spotify: {
-		name: 'Spotify',
-		icon: SpotifyIcon,
-	},
-	tumblr: {
-		name: 'Tumblr',
-		icon: TumblrIcon,
-	},
-	twitch: {
-		name: 'Twitch',
-		icon: TwitchIcon,
-	},
-	twitter: {
-		name: 'Twitter',
-		icon: TwitterIcon,
-	},
-	vimeo: {
-		name: 'Vimeo',
-		icon: VimeoIcon,
-	},
-	vk: {
-		name: 'VK',
-		icon: VkIcon,
-	},
-	wordpress: {
-		name: 'WordPress',
-		icon: WordPressIcon,
-	},
-	yelp: {
-		name: 'Yelp',
-		icon: YelpIcon,
-	},
-	youtube: {
-		name: 'YouTube',
-		icon: YouTubeIcon,
-	},
-};
+import { find } from 'lodash';
 
-export default socialList;
+/**
+ * Internal dependencies
+ */
+import variations from './variations';
 
 /**
  * Retrieves the social service's icon component.
  *
- * @param {string} site key for a social service (lowercase slug)
+ * @param {string} name key for a social service (lowercase slug)
  *
  * @return {WPComponent} Icon component for social service.
  */
-export const getIconBySite = ( site ) => {
-	return socialList[ site ].icon;
+export const getIconBySite = ( name ) => {
+	return find( variations, { name } ).icon;
```

Comment:
It needs to be guarded in the case where a custom variation is registered and the site can't be found.